plugins/meta/firewall/README.md

```diff
@@ -133,3 +133,99 @@ of the container as shown:
+`nftables` backend might look like:
+
+```json
+{
+   "cniVersion": "0.4.0",
+   "name": "podman",
+   "plugins": [
+      {
+         "type": "bridge",
+         "bridge": "cni-podman0",
+         "isGateway": true,
+         "ipMasq": true,
+         "ipam": {
+            "type": "host-local",
+            "routes": [
+               {
+                  "dst": "0.0.0.0/0"
+               }
+            ],
+            "ranges": [
+               [
+                  {
+                     "subnet": "192.168.124.0/24",
+                     "gateway": "192.168.124.1"
+                  }
+               ]
+            ]
+         }
+      },
+      {
+         "type": "portmap",
+         "capabilities": {
+            "portMappings": true
+         }
+      },
+      {
+         "type": "firewall",
+         "backend": "nftables"
+      }
+   ]
+}
+```
+
+Prior to the invocation of CNI `firewall` plugin, the `FORWARD` chain in `filter` table is:
+
+```
+$ nft list chain ip filter FORWARD -a
+table ip filter {
+        chain FORWARD { # handle 2
+                type filter hook forward priority 0; policy drop;
+                oifname "virbr0" ip daddr 192.168.122.0/24 ct state established,related counter packets 0 bytes 0 accept # handle 51
+                iifname "virbr0" ip saddr 192.168.122.0/24 counter packets 0 bytes 0 accept # handle 52
+                iifname "virbr0" oifname "virbr0" counter packets 0 bytes 0 accept # handle 53
+                log prefix "IPv4 FORWARD drop: " flags all # handle 54
+                counter packets 10 bytes 630 drop # handle 55
+        }
+}
+```
+
+After starting a container, the plugin executes the following commands based
+on the configuration above. Please note that `position 51` refers to the handle
+at the top of the chain.
+
+```
+nft insert rule filter FORWARD position 51 oifname "cni-podman0" ip daddr 192.168.124.0/24 ct state established,related counter packets 0 bytes 0 accept
+nft insert rule filter FORWARD position 51 iifname "cni-podman0" ip saddr 192.168.124.0/24 counter packets 0 bytes 0 accept
+nft insert rule filter FORWARD position 51 iifname "cni-podman0" oifname "cni-podman0" counter packets 0 bytes 0 accept
+```
+
+After the plugin's execution, the chain looks like this:
+
+```
+$ nft list chain ip filter FORWARD -a
+table ip filter {
+        chain FORWARD { # handle 2
+                type filter hook forward priority 0; policy drop;
+                oifname "cni-podman0" ip daddr 192.168.124.0/24 ct state established,related counter packets 100 bytes 113413 accept # handle 71
+                iifname "cni-podman0" ip saddr 192.168.124.0/24 counter packets 124 bytes 12996 accept # handle 72
+                iifname "cni-podman0" oifname "cni-podman0" counter packets 0 bytes 0 accept # handle 73
+                oifname "virbr0" ip daddr 192.168.122.0/24 ct state established,related counter packets 0 bytes 0 accept # handle 51
```

Comment:
How about move these rules into a new chain of filter table? just like `iptables` backend

Reply:
> How about move these rules into a new chain of filter table? just like iptables backend

@mars1024 , the challenge with doing that is you end up having a lot of chain that you would need to pull. It is like pulling the entire config. In case of `iptables` backend, there is a good library behind it, i.e. `go-iptables`. In the case of `nftables`, there is `google/nftables`, but it is not as extensible as it comes to rule. see https://github.com/sbezverk/nftableslib/issues/56#issuecomment-599162875

Here, we list a single chain, find the rules with relevant interfaces, and add/remove what is no longer necessary.

Reply:
> How about move these rules into a new chain of filter table? just like iptables backend

@mars1024 , creating separate chains with a jump in the FORWARD chain.